internal/config/endpoints.go

```diff
@@ -25,176 +35,141 @@ type Endpoint struct {
-	"northamerica-northeast2": {
-		URL:        "https://northamerica-northeast2-5tkroniexa-pd.a.run.app",
-		Region:     "northamerica-northeast2",
-		RegionName: "Toronto",
-	},
-	"southamerica-east1": {
-		URL:        "https://southamerica-east1-5tkroniexa-rj.a.run.app",
-		Region:     "southamerica-east1",
-		RegionName: "São Paulo",
-	},
-	"southamerica-west1": {
-		URL:        "https://southamerica-west1-5tkroniexa-tl.a.run.app",
-		Region:     "southamerica-west1",
-		RegionName: "Santiago",
-	},
-	"us-central1": {
-		URL:        "https://us-central1-5tkroniexa-uc.a.run.app",
-		Region:     "us-central1",
-		RegionName: "Iowa",
-	},
-	"us-east1": {
-		URL:        "https://us-east1-5tkroniexa-ue.a.run.app",
-		Region:     "us-east1",
-		RegionName: "South Carolina",
-	},
-	"us-east4": {
-		URL:        "https://us-east4-5tkroniexa-uk.a.run.app",
-		Region:     "us-east4",
-		RegionName: "North Virginia",
-	},
-	"us-east5": {
-		URL:        "https://us-east5-5tkroniexa-ul.a.run.app",
-		Region:     "us-east5",
-		RegionName: "Columbus",
-	},
-	"us-west1": {
-		URL:        "https://us-west1-5tkroniexa-uw.a.run.app",
-		Region:     "us-west1",
-		RegionName: "Oregon",
-	},
-	"us-west2": {
-		URL:        "https://us-west2-5tkroniexa-wl.a.run.app",
-		Region:     "us-west2",
-		RegionName: "Los Angeles",
-	},
-	"us-west3": {
-		URL:        "https://us-west3-5tkroniexa-wm.a.run.app",
-		Region:     "us-west3",
-		RegionName: "Salt Lake City",
-	},
-	"us-west4": {
-		URL:        "https://us-west4-5tkroniexa-wn.a.run.app",
-		Region:     "us-west4",
-		RegionName: "Las Vegas",
-	},
+type innerEndpoint struct {
+	//Used to unmarshal properly formatted JSON obtained from /api/endpoint JSON
+	URL        string
+	Region     string
+	RegionName string
+}
+
+var allEndpoints map[string]Endpoint
+
+func getJson(url string, target interface{}) error {
+	r, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer r.Body.Close()
+	return json.NewDecoder(r.Body).Decode(target)
+}
+
+func GetEndpoints() map[string]Endpoint {
+	//TODO detect when running in Cloud Run and GenerateConfigFromAPI()
+	if len(allEndpoints) == 0 {
+		allEndpoints = GenerateConfigFromEndpoints()
+	}
+	return allEndpoints
+}
+
+func GenerateConfigFromEndpoints() map[string]Endpoint {
+	//Used by CLI to pull endpoint configs from Cloud Run enpoints.
+	e := make(map[string]Endpoint)
+	ie := new(map[string]innerEndpoint)
+	getJson("https://global.gcping.com/api/endpoints", &ie)
+	for _, s := range *ie {
+		e[s.Region] = Endpoint(s)
+	}
+	return e
+}
+
+func GenerateConfigFromAPI(ctx context.Context) map[string]Endpoint {
+	//Used by Cloud Run Endpoints to pull endpoint configs from Cloud Run Admin API
+	log.Print("Using Cloud Run Admin API to generate Endpoints config.")
+	runService, err := run.NewService(ctx)
+	// TODO: Get project name from Cloud Run metadata service if not defined in env variable
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+
+	if err != nil {
+		panic(err)
```

Comment:
Not sure why I didn't catch this earlier. 

It is usually not great practice to `panic`. Would rather have the return values be `(map[string]Endpoint, error)` and `return nil, err` if an error occurs. (similarly for other `panic`s)